encode.go

```diff
@@ -11,74 +11,92 @@ import (
 )
 
 // TODO: Handle all the errors
-func writeData(dbuf *bytes.Buffer, order binary.ByteOrder, data *K) error {
-	binary.Write(dbuf, order, data.Type)
-	if K0 <= data.Type && data.Type <= KT { // All Lists
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(reflect.ValueOf(data.Data).Len()))
-	} else if data.Type == XT { // Table
-		binary.Write(dbuf, order, data.Attr)
+func write(buf *bytes.Buffer, order binary.ByteOrder, k *K) error {
+	binary.Write(buf, order, k.Type)
+	if K0 <= k.Type && k.Type <= KT { // All Lists
+		binary.Write(buf, order, k.Attr)
+		binary.Write(buf, order, int32(reflect.ValueOf(k.Data).Len()))
+	} else if k.Type == XT { // Table
+		binary.Write(buf, order, k.Attr)
 	}
 
-	switch data.Type {
+	switch k.Type {
 	case K0:
-		for _, toSend := range data.Data.([]*K) {
-			err := writeData(dbuf, order, toSend)
+		for _, toSend := range k.Data.([]*K) {
+			err := write(buf, order, toSend)
 			if err != nil {
 				return err
 			}
 		}
-	case -KB, -UU, -KG, -KH, -KI, -KJ, -KE, -KF, -KC, -KD, -KT, // TODO: case: -KM , -KZ, -KN, -KU, -KV
-		KB, UU, KG, KH, KI, KJ, KE, KF, KM, KD, KZ, KN, KU, KV, KT:
+	case -KB, -UU, -KG, -KH, -KI, -KJ, -KE, -KF, -KC, -KM , -KZ, -KN, -KU, -KV,
+		+KB, +UU, +KG, +KH, +KI, +KJ, +KE, +KF, +KM, +KZ, +KN, +KU, +KV:
 		// Scalar & Vector of boolean, int variants, float variant, and bytes.
-		// Note: UUID is backed by byte array of length 16
-		binary.Write(dbuf, order, data.Data)
+		// Note: UUID is back by byte array of length 16
```

Comment:
(nit) backed

Reply:
I don't know how this happened. Will update